src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts

```diff
@@ -44,30 +57,41 @@ export class GenerateLetterComponent implements OnInit {
 
   get reasons(): Array<FormNodeOption<string>> {
     return [
-      { label: 'Authorised', value: LettersOfAuth.LetterTypeEnum.Authorization },
-      { label: 'Rejected', value: LettersOfAuth.LetterTypeEnum.Rejection }
+      { label: 'Trailer authorised', value: LETTER_TYPES[0].value },
+      { label: 'Trailer rejected', value: LETTER_TYPES[1].value }
     ];
   }
 
-  ngOnInit(): void {
-    if (!this.currentTechRecord || this.currentTechRecord.vehicleType !== 'trl') {
-      this.navigateBack();
-    }
-
-    this.actions$.pipe(ofType(generateLetterSuccess), take(1)).subscribe(() => this.navigateBack());
+  get mostRecentLetter(): LettersOfAuth | undefined {
+    return (
+      this.currentTechRecord &&
+      cloneDeep(this.currentTechRecord.lettersOfAuth)
```

Comment:
lettersOfAuth is an object, not an array, you will only ever have the most recent

Reply:
It's currently stored as an array in VTM but I can make that change